Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.cpp

```diff
@@ -540,192 +485,202 @@ ActsExamples::ProcessCode ActsExamples::VertexPerformanceWriter::writeT(
-            m_recoZ.push_back(vtx.fullPosition()[Acts::FreeIndices::eFreePos2]);
-            m_recoT.push_back(vtx.fullPosition()[Acts::FreeIndices::eFreeTime]);
-
-            const Acts::ActsVector<4> diffPos = vtx.fullPosition() - truePos;
-            m_resX.push_back(diffPos[Acts::FreeIndices::eFreePos0]);
-            m_resY.push_back(diffPos[Acts::FreeIndices::eFreePos1]);
-            m_resZ.push_back(diffPos[Acts::FreeIndices::eFreePos2]);
-            m_resT.push_back(diffPos[Acts::FreeIndices::eFreeTime]);
-
-            Acts::ActsScalar varX = vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos0, Acts::FreeIndices::eFreePos0);
-            Acts::ActsScalar varY = vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos1, Acts::FreeIndices::eFreePos1);
-            Acts::ActsScalar varZ = vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos2, Acts::FreeIndices::eFreePos2);
-            Acts::ActsScalar varTime = vtx.fullCovariance()(
-                Acts::FreeIndices::eFreeTime, Acts::FreeIndices::eFreeTime);
-            m_pullX.push_back(
-                pull(diffPos[Acts::FreeIndices::eFreePos0], varX, "X"));
-            m_pullY.push_back(
-                pull(diffPos[Acts::FreeIndices::eFreePos1], varY, "Y"));
-            m_pullZ.push_back(
-                pull(diffPos[Acts::FreeIndices::eFreePos2], varZ, "Z"));
-            m_pullT.push_back(
-                pull(diffPos[Acts::FreeIndices::eFreeTime], varTime, "T"));
-
-            m_covXX.push_back(varX);
-            m_covYY.push_back(varY);
-            m_covZZ.push_back(varZ);
-            m_covTT.push_back(varTime);
-            m_covXY.push_back(vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos0, Acts::FreeIndices::eFreePos1));
-            m_covXZ.push_back(vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos0, Acts::FreeIndices::eFreePos2));
-            m_covXT.push_back(vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos0, Acts::FreeIndices::eFreeTime));
-            m_covYZ.push_back(vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos1, Acts::FreeIndices::eFreePos2));
-            m_covYT.push_back(vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos1, Acts::FreeIndices::eFreeTime));
-            m_covZT.push_back(vtx.fullCovariance()(
-                Acts::FreeIndices::eFreePos2, Acts::FreeIndices::eFreeTime));
-
-            double sumPt2 = 0;
-            for (const auto& trk : tracksAtVtx) {
-              if (trk.trackWeight > m_cfg.minTrkWeight) {
-                double pt = trk.originalParams.as<Acts::BoundTrackParameters>()
-                                ->transverseMomentum();
-                sumPt2 += pt * pt;
-              }
-            }
-            m_sumPt2.push_back(sumPt2);
+      const auto& truthVertexParticle =
+          *std::find_if(particleAtVtx.begin(), particleAtVtx.end(),
```

Comment:
This is guaranteed never to return `end()`?

Reply:
I think so yes because we can only match this vertex if there was at least one particle at the vertex